app/src/main/java/io/gnosis/safe/ui/settings/owner/ledger/transport/LedgerWrapper.kt

```diff
@@ -0,0 +1,65 @@
+package io.gnosis.safe.ui.settings.owner.ledger
+
+import io.gnosis.safe.ui.settings.owner.ledger.transport.LedgerException
+import io.gnosis.safe.ui.settings.owner.ledger.transport.SerializeHelper
+import java.io.ByteArrayOutputStream
+
+
+object LedgerWrapper {
+
+    private const val TAG_APDU = 0x05
+
+    fun wrapAPDU(data: ByteArray): ByteArray {
+        val apdu = ByteArrayOutputStream()
+        apdu.write(TAG_APDU)
+        apdu.write(0x00)
+        apdu.write(0x00)
+        SerializeHelper.writeUint16BE(apdu, data.count().toLong())
+        apdu.write(data)
+        return apdu.toByteArray()
+    }
+
+    fun unwrapAPDU(data: ByteArray): ByteArray {
+        if (data.size <= 6 || data[0] != 0x05.toByte()) {
+            throw LedgerException(LedgerException.ExceptionReason.IO_ERROR, "invalid data size")
```

Comment:
Since this is not only about the size. Wouldn't `invalid data format `be more precise?